Shared/Store/BaseDataStore.swift

```diff
@@ -0,0 +1,438 @@
+        //          * Anything to Syncing at the start of sync after the first syncing starts
+        //          * Syncing to Synced after all syncs.
+        //
+        //      (in sync world email confirmation happens as part of sync, and sync end happens even if not confirmed).
+        switch (storageState, syncState, notification.name) {
+        case (.Unprepared, _, NotificationNames.ProfileDidStartSyncing):
+            syncSubject.onNext(.Syncing)
+        case (_, _, NotificationNames.ProfileDidStartSyncing):
+            // fall through for the locked and unlocked states.
+            syncSubject.onNext(.Syncing)
+        case (_, _, NotificationNames.ProfileDidFinishSyncing):
+            // end of all syncs
+            syncSubject.onNext(.Synced)
+        default:
+            print("Unexpected state combination: \(storageState) | \(syncState), \(notification.name)")
+        }
+    }
+
+    private func updateList() {
+        let logins = self.profile.logins
+        logins.getAllLogins() >>== { (cursor: Cursor<Login>) in
+            self.listSubject.accept(cursor.asArray())
+        }
+    }
+
+    private func makeEmptyList() {
+        self.listSubject.accept([])
+    }
+}
+
+extension BaseDataStore {
+    public func remove(id: String) {
+        self.profile.logins.removeLoginByGUID(id) >>== {
+            self.syncSubject.onNext(.ReadyToSync)
+        }
+    }
+
+    public func add(item: LoginData) {
+        self.profile.logins.addLogin(item) >>== {
+            self.syncSubject.onNext(.ReadyToSync)
+        }
+    }
+}
+
+// initialization
+extension BaseDataStore {
+    private func initializeProfile() {
+        self.profile.syncManager?.applicationDidBecomeActive()
+
+        self.fxaLoginHelper.application(UIApplication.shared, didLoadProfile: profile)
+    }
+
+    private func setInitialState() {
+        guard profile.hasSyncableAccount() else {
+            if !profile.hasAccount() {
+                // first run.
+                self.storageStateSubject.onNext(.Unprepared)
+            }
+
+            self.syncSubject.onNext(.NotSyncable)
+            return
+        }
+        self.syncSubject.onNext(.ReadyToSync)
+        self.handleLock()
+    }
+
+    private func handleLock() {
+        // default to locked state
+        self.storageStateSubject.onNext(.Locked)
+
+        self.userDefaults.onAutoLockTime
+                .take(1)
+                .subscribe(onNext: { autoLockSetting in
+                    switch autoLockSetting {
+                    case .Never:
+                        self.unlock()
+                    default:
+                        let date = NSDate(
+                                timeIntervalSince1970: self.userDefaults.double(
+                                        forKey: UserDefaultKey.autoLockTimerDate.rawValue))
+
+                        if date.timeIntervalSince1970 > 0 && date.timeIntervalSinceNow > 0 {
```

Comment:
Prefer to remove `>`.

oh, wait this is to make sure we have an `autoLockTimerDate`.

```swift
if let seconds = self.userDefaults.double(forKey: UserDefaultKey.autoLockTimerDate.rawValue),
     seconds > 0 {

} else {

}
```

Actually, I'm not sure what this code is supposed to do, or that it wasn't looked at in another PR.

Reply:
this is from another PR.